hyperspy/_components/lorentzian.py

```diff
@@ -16,76 +16,82 @@
+    :math:`\gamma`  gamma  
+    :math:`x_0`     centre 
+    ============== =============
+
+
+    Parameters
+    -----------
+    A : float
+        Height parameter, where :math:`A/(\gamma\pi)` is the maximum of the 
+        peak.
+    gamma : float
+        Scale parameter corresponding to the half-width-at-half-maximum of the 
+        peak, which corresponds to the interquartile spread.
+    centre : float
+        Location of the peak maximum.
+    **kwargs
+        Extra keyword arguments are passed to the ``Expression`` component.
+
+
+    For convenience the `fwhm` attribute can be used to get and set
+    the full-with-half-maximum.
     """
 
-    def __init__(self, A=1., gamma=1., centre=0.):
-        Component.__init__(self, ('A', 'gamma', 'centre'))
-        self.A.value = A
-        self.gamma.value = gamma
-        self.centre.value = centre
+    def __init__(self, A=1., gamma=1., centre=0., module="numexpr", **kwargs):
+        # We use `_gamma` internally to workaround the use of the `gamma` 
+        # function in sympy
+        super(Lorentzian, self).__init__(
+            expression="A / pi * (_gamma / ((x - centre)**2 + _gamma**2))",
+            name="Lorentzian",
+            A=A,
+            _gamma=gamma,
+            centre=centre,
+            position="centre",
+            module=module,
+            autodoc=False,
+            **kwargs)
 
         # Boundaries
         self.A.bmin = 0.
         self.A.bmax = None
-        self.gamma.bmin = None
-        self.gamma.bmax = None
-        self._position = self.centre
+
+        self._gamma.bmin = 0.
+        self._gamma.bmax = None
+        self._gamma.name = 'gamma'
 
         self.isbackground = False
         self.convolved = True
 
-        # Gradients
-        self.A.grad = self.grad_A
-        self.gamma.grad = self.grad_gamma
-        self.centre.grad = self.grad_centre
-
-    def function(self, x):
-        """
-        """
-        A = self.A.value
-        gamma = self.gamma.value
-        centre = self.centre.value
-
-        return A / np.pi * (gamma / ((x - centre) ** 2 + gamma ** 2))
-
-    def grad_A(self, x):
-        """
-        """
-        return self.function(x) / self.A.value
-
-    def grad_gamma(self, x):
-        """
-        """
-        return self.A.value / (np.pi * (self.gamma.value ** 2 + (x - self.centre.value) ** 2)) - (
-            (2 * self.A.value * self.gamma.value ** 2) / (np.pi * (self.gamma.value ** 2 + (x - self.centre.value) ** 2) ** 2))
-
-    def grad_centre(self, x):
-        """
-        """
-        return (2 * (x - self.centre.value) * self.A.value * self.gamma.value) / \
-            (np.pi *
-             (self.gamma.value ** 2 + (x - self.centre.value) ** 2) ** 2)
+    @property
+    def fwhm(self):
+        return self._gamma.value * 2
+
+    @fwhm.setter
+    def fwhm(self, value):
+        self._gamma.value = value / 2
+
+    @property
+    def gamma(self):
+        return self._gamma
+
+    @gamma.setter
```

Comment:
I think that this property setter is unnecessary because users don't need to set the gamma parameter.

Reply:
We need the setter to avoid changing the API.

Reply:
Can you provide more details?

Reply:
See https://github.com/hyperspy/hyperspy/pull/2134#issuecomment-477182152 and related discussion.
In short: `gamma` can't be used as a parameter because `sympy` already has it defined as a function. @jlaehne used `_gamma` as workaround and I suggested the property/setter to keep the same API.

Reply:
I see, but I am not proposing to remove the property, just the property setter because I don't see why somebody would like to replace the ``gamma`` parameter by something else.

Reply:
Well, using the property setter is a possible way to set starting parameters prior to fitting.

Reply:
I think that it is not necessary, because the setter is to set the ``gamma`` attribute itself. To set the starting parameters you use the ``value`` attribute of the ``gamma`` ``Parameter``.

Reply:
> I see, but I am not proposing to remove the property, just the property setter because I don't see why somebody would like to replace the `gamma` parameter by something else.

Yes, I misread your first comment. In this case, they should not be a need to set `gamma` since during the initialisation of the component, `_gamma` is being set!

Reply:
OK, your right. I tested again and the `@property` alone does the job, the setter is superfluous, so I removed it.